app/lib/submission_builder/return_header.rb

```diff
@@ -12,6 +12,12 @@ def document
           xml.TaxPeriodEndDt date_type(Date.new(@submission.data_source.tax_return_year, 12, 31))
         end
         xml.TaxYr @submission.data_source.tax_return_year
+        if @submission.data_source.state_code.upcase == "NJ"
```

Comment:
[sand] if we don't have a better way of disambiguating state-specific blocks, we should make one.